src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,392 @@
+'''
+A module to load annotation files for Sentinel-1 IW SLC SAFE data
+To be used for the class "Sentinel1BurstSlc"
+'''
+
+from dataclasses import dataclass
+import datetime
+import xml.etree.ElementTree as ET
+
+import numpy as np
+from scipy.interpolate import InterpolatedUnivariateSpline
+
+@dataclass
+class AnnotationBase:
+    '''
+    A virtual base class of the inheriting annotation class i.e. Product, Calibration, and Noise.
+    Not intended for standalone use.
+    '''
+    # A parent class for annotation reader for Calibrarion, Noise, and Product
+    xml_et: ET
+    #kind:str
+
+    @classmethod
+    def _parse_scalar(cls, path_field: str, str_type: str):
+        '''Parse the scalar value in the annotation
+        path_field: field in the xml_et to parse
+        str_type: Choose how the taxts in the field will be parsed: {datetime, scalar_int, scalar_float, vector_int, vector_float, str}
```

Comment:
My apologies for not being specific enough in my documentation request. Could you please format this according to the [numpy documentation standard](https://numpydoc.readthedocs.io/en/latest/format.html)? (It's a standard we've carried over from ISCE3).

Here's an example to follow from `s1_orbit.py`:
https://github.com/opera-adt/s1-reader/blob/b6f0400b4a77d02a2b9beca85d8c5d6d1d841980/src/s1reader/s1_orbit.py#L183-L199